tests/common.py

```diff
@@ -392,45 +406,59 @@ def assert_equal_vectors(self, truth, test, rtol):
             assert 0
 
     def compare_forces(
-        self, x, params, box, lamb, ref_potential, test_potential, rtol, precision, atol=1e-8, benchmark=False
+        self,
+        x: NDArray,
+        params: NDArray,
+        box: NDArray,
+        lambdas: Union[List[float], float],
+        ref_potential,
+        test_potential,
+        rtol: float,
+        precision,
+        atol: float = 1e-8,
+        benchmark: bool = False,
     ):
 
         test_impl = test_potential.unbound_impl(precision)
 
         x = (x.astype(np.float32)).astype(np.float64)
         params = (params.astype(np.float32)).astype(np.float64)
 
+        if isinstance(lambdas, float):
+            lambdas = [lambdas]
+
         assert x.ndim == 2
         # N = x.shape[0]
         # D = x.shape[1]
 
         assert x.dtype == np.float64
         assert params.dtype == np.float64
 
-        ref_u = ref_potential(x, params, box, lamb)
-        grad_fn = jax.grad(ref_potential, argnums=(0, 1, 3))
-        ref_du_dx, ref_du_dp, ref_du_dl = grad_fn(x, params, box, lamb)
-
-        for combo in itertools.product([False, True], repeat=4):
-
-            (compute_du_dx, compute_du_dp, compute_du_dl, compute_u) = combo
-
-            # do each computation twice to check determinism
-            test_du_dx, test_du_dp, test_du_dl, test_u = test_impl.execute_selective(
-                x, params, box, lamb, compute_du_dx, compute_du_dp, compute_du_dl, compute_u
-            )
-            if compute_u:
-                np.testing.assert_allclose(ref_u, test_u, rtol=rtol, atol=atol)
-            if compute_du_dx:
-                self.assert_equal_vectors(np.array(ref_du_dx), np.array(test_du_dx), rtol)
-            if compute_du_dl:
-                np.testing.assert_allclose(ref_du_dl, test_du_dl, rtol=rtol)
-            if compute_du_dp:
-                np.testing.assert_allclose(ref_du_dp, test_du_dp, rtol=rtol, atol=atol)
-
-            test_du_dx_2, test_du_dp_2, test_du_dl_2, test_u_2 = test_impl.execute_selective(
-                x, params, box, lamb, compute_du_dx, compute_du_dp, compute_du_dl, compute_u
-            )
+        for lamb in lambdas:
```

Comment:
should we add a note on why it's preferable to move the loop over lambda inside `compare_forces`?